tests/cppunit/types/json_test.cc

```diff
@@ -262,7 +262,9 @@ TEST_F(RedisJsonTest, ArrLen) {
   ASSERT_EQ(res.size(), 1);
   ASSERT_EQ(res[0], 3);
   res.clear();
-  // 5. path is not found
+  // 5. key/path is not found
+  ASSERT_FALSE(json_->ArrLen("not_exists", "$.*", res).ok());
+  ASSERT_TRUE(res.empty());
```

Comment:
We don't need to check it here. I think if the result is not `ok()`, the result can be undefined(Though it's empty here)

Reply:
I saw that many functions through the codebase explicitly clear the output parameter at the beginning of the function. Can we do that inside Json::ArrLen()? I mean, call `arr_lens->clear()`. What do you think, @mapleFU @PragmaTwice ?

Reply:
I'm +1 for this.

Reply:
Yeah. And I'm planning replace these out pointers with StatusOr to eliminate such issues.

Reply:
`StatusOr<vector<>>` looks great! 🍻